libkrun/libkruntime.c

```diff
@@ -35,44 +40,115 @@ clock_gettime_monotonic()
  * JNI Implementation -- Optionally compiled in
  */
 #ifdef WITH_JAVA
-#include <jni.h>
 
 JNIEXPORT jdouble JNICALL
 Java_IterationsRunner_JNI_1clock_1gettime_1monotonic(JNIEnv *e, jclass c) {
     return (jdouble) clock_gettime_monotonic();
 }
 
 JNIEXPORT jlong JNICALL
-Java_IterationsRunner_JNI_1read_1ts_1reg(JNIEnv *e, jclass c)
+Java_IterationsRunner_JNI_1read_1ts_1reg_1start(JNIEnv *e, jclass c)
+{
+    return read_ts_reg_start();
+}
+
+JNIEXPORT jlong JNICALL
+Java_IterationsRunner_JNI_1read_1ts_1reg_1stop(JNIEnv *e, jclass c)
 {
-    return read_ts_reg();
+    return read_ts_reg_stop();
 }
 #endif
 
 /*
  * Support for reading the TS (timestamp) register.
  *
- * This assumes and x86 CPU.
+ * These functions assume an x86-64 CPU.
+ *
+ * Fore more info, see page 16 of:
  *
- * Derived from:
- * $OpenBSD: pctr.h,v 1.5 2014/03/29 18:09:28 guenther Exp $
+ * "How to Benchmark Code Execution Times on Intel® IA-32 and IA-64
+ * Instruction Set Architectures"
```

Comment:
I think we should add something like "by Gabriele Paoloni" (it might help future generations find a copy of the PDF).